CppCoreGuidelines.md

```diff
@@ -8009,35 +7992,20 @@ In this example, `Shape` does not inherit from `Geometric_attributes`. Only its
     
 ##### Example
     
-    class Base {
-    public:
-        virtual void f() { }
-    };
-    
-    class Derived : public Base {
-    public:
-        void f() override { }
-    };
-    
-    class NotDerivedFromBase {
-    public:
-        virtual void f() { }
-    };
+    class A { ... };
     
-    class DerivedFromNotDerivedFromBaseAndFromBase : public NotDerivedFromBase, public Base {
-        void f() override { }
-    };
+    class B { ... };
     
-    void some_function(NotDerivedFromBase* p) {
-      auto base = dynamic_cast<Base*>(p);
-        if (base == nullptr) {
-          //...
-        }
+    void some_function(B* p) {
+      auto base = dynamic_cast<A*>(p);
+      if (base == nullptr) {
+        // check first, before deref
```

Comment:
Should this be `base != nullref`? To me the comment suggests it's going to be dereferenced inside the block guarded by the null check.